Core/BookmarksCachingSearch.swift

```diff
@@ -204,7 +204,7 @@ public class BookmarksCachingSearch: BookmarksStringSearch {
             let title = entry.title.lowercased()
 
             // Exact matches - full query
-            if title.starts(with: query) { // High score for exact match from the beginning of the title
+            if title.trimmingCharacters(in: .alphanumerics.inverted).starts(with: query) { // High score for exact match from the beginning of the title
```

Comment:
Unfortunately Bookmarks scoring doesn't re-use the same code as Suggestions so had to reapply the fix also applied to BSK here too.